compiler/rustc_arena/src/lib.rs

```diff
@@ -48,17 +43,16 @@ pub struct TypedArena<T> {
     _own: PhantomData<T>,
 }
 
+#[repr(transparent)]
 struct TypedArenaChunk<T> {
     /// The raw storage for the arena chunk.
-    storage: RawVec<T>,
-    /// The number of valid entries in the chunk.
-    entries: usize,
+    inner: Vec<T>,
```

Comment:
This feels wrong to do without a major rewrite of the entire thing. Now there are workarounds around the implicit behaviour of `Vec`. I think we should not do this commit without rewriting the entire code to properly use `Vec` (so no more `set_len` either).

Reply:
I don't want to touch *too* much because I fear the potential impacts on performance etc. this might have. Better be careful :). But if it helps I'll remove the `TypedArenaChunk` type and replace it with `Vec` only.

Reply:
I don't think we should be touching this at all tbh, all the other changes in this PR lgtm

Reply:
@oli-obk ok I'll rework it and move it to a different PR.